prog/vm/github_runner.rb

```diff
@@ -177,16 +194,20 @@ def setup_info
 
       # We placed the script in the "/usr/local/share/" directory while generating
       # the golden image. However, it needs to be moved to the home directory because
-      # the runner creates some configuration files at the script location. The "runner"
-      # user doesn't have write permission for the "/usr/local/share/" directory.
+      # the runner creates some configuration files at the script location. Since the
+      # github runner vm is created with the runneradmin user, directory is first moved
+      # to runneradmin user's home directory. At the end of this script, it will be moved
+      # to runner user's home folder.
       sudo [ ! -d /usr/local/share/actions-runner ] || sudo mv /usr/local/share/actions-runner ./
-      sudo chown -R runner:runner actions-runner
+      sudo [ ! -d /home/runner/actions-runner ] || sudo mv /home/runner/actions-runner ./
```

Comment:
Why don't we combine this line with the one above? Is it because of idempotency? If that's the case, could you please add a comment explaining it? I'm certain that in the future, we might forget and attempt to refactor and merge them.

Reply:
Yes, it is for idempotency purpose to make sure that we can continue if the script fails after moving actions-runner to ./ either from /usr/local/share/actions-runner or /home/runner